compiler/scripts/anonymize.js

```diff
@@ -0,0 +1,171 @@
+ * LICENSE file in the root directory of this source tree.
+ */
+
+'use strict';
+
+const fs = require('fs');
+const HermesParser = require('hermes-parser');
+const BabelParser = require('@babel/parser');
+const BabelCore = require('@babel/core');
+const invariant = require('invariant');
+const {argv, stdin} = require('process');
+const prettier = require('prettier');
+const {JSXText} = require('hermes-parser/dist/generated/ESTreeVisitorKeys');
+
+function runPlugin(text, file, language) {
+  let ast;
+  if (language === 'flow') {
+    ast = HermesParser.parse(text, {
+      babel: true,
+      flow: 'all',
+      sourceFilename: file,
+      sourceType: 'module',
+      enableExperimentalComponentSyntax: true,
+    });
+  } else {
+    ast = BabelParser.parse(text, {
+      sourceFilename: file,
+      plugins: ['typescript', 'jsx'],
+      sourceType: 'module',
+    });
+  }
+  const result = BabelCore.transformFromAstSync(ast, text, {
+    ast: false,
+    filename: file,
+    highlightCode: false,
+    retainLines: true,
+    plugins: [[AnonymizePlugin]],
+    sourceType: 'module',
+    configFile: false,
+    babelrc: false,
+  });
+  invariant(
+    result?.code != null,
+    `Expected BabelPluginReactForget to codegen successfully, got: ${result}`
+  );
+  return result.code;
+}
+
+async function format(code, language) {
+  return await prettier.format(code, {
+    semi: true,
+    parser: language === 'typescript' ? 'babel-ts' : 'flow',
+  });
+}
+
+const TAG_NAMES = new Set([
+  'div',
+  'span',
+  'input',
+  'button',
+  'a',
+  'form',
+  'select',
+  'textarea',
+  'body',
+  'head',
+  'html',
+]);
+
+function AnonymizePlugin(_babel) {
+  let index = 0;
+  const identifiers = new Map();
+  const literals = new Map();
+  return {
+    name: 'anonymize',
+    visitor: {
+      JSXNamespacedName(path) {
+        throw error('TODO: handle JSXNamedspacedName');
+      },
+      JSXIdentifier(path) {
+        const name = path.node.name;
+        let nextName = identifiers.get(name);
+        if (nextName == null) {
+          const isCapitalized =
+            name.slice(0, 1).toUpperCase() === name.slice(0, 1);
+          nextName = isCapitalized
+            ? `Component${(index++).toString(16).toUpperCase()}`
+            : TAG_NAMES.has(name)
+              ? name
+              : `c${(index++).toString(16)}`;
+          identifiers.set(name, nextName);
+        }
+        path.node.name = nextName;
+      },
+      Identifier(path) {
+        const name = path.node.name;
+        let nextName = identifiers.get(name);
+        if (nextName == null) {
+          const isCapitalized =
+            name.slice(0, 1).toUpperCase() === name.slice(0, 1);
+          const prefix = isCapitalized ? 'V' : 'v';
+          nextName = `${prefix}${(index++).toString(16)}`;
+          if (name.startsWith('use')) {
+            nextName =
+              'use' + nextName.slice(0, 1).toUpperCase() + nextName.slice(1);
```

Comment:
May wish to retain builtin hook calls, since the compiler will behave differently on `useMemo`, `useEffect`, than on `useInternalMetaHook`